lib/resourcemerge/apiext.go

```diff
@@ -12,18 +12,36 @@ import (
 // modified is set to true when existing had to be updated with required.
 func EnsureCustomResourceDefinitionV1(modified *bool, existing *apiextv1.CustomResourceDefinition, required apiextv1.CustomResourceDefinition) {
 	EnsureObjectMeta(modified, &existing.ObjectMeta, required.ObjectMeta)
+	ensureCustomResourceDefinitionV1Defaults(&required)
+	ensureCustomResourceDefinitionV1CaBundle(&required, *existing)
+	if !equality.Semantic.DeepEqual(existing.Spec, required.Spec) {
+		*modified = true
+		existing.Spec = required.Spec
+	}
+}
 
-	// apply defaults to blue print
+func ensureCustomResourceDefinitionV1Defaults(required *apiextv1.CustomResourceDefinition) {
 	if len(required.Spec.Names.Singular) == 0 {
 		required.Spec.Names.Singular = strings.ToLower(required.Spec.Names.Kind)
 	}
 	if len(required.Spec.Names.ListKind) == 0 {
 		required.Spec.Names.ListKind = required.Spec.Names.Kind + "List"
 	}
+}
 
-	// we stomp everything
-	if !equality.Semantic.DeepEqual(existing.Spec, required.Spec) {
-		*modified = true
-		existing.Spec = required.Spec
+// ensureCustomResourceDefinitionV1CaBundle ensures that the field
+// .Spec.Conversion.Webhook.ClientConfig.CABundle is not managed by the CVO when
+// "service.beta.openshift.io/inject-cabundle=true" annotation is set in existing.
+func ensureCustomResourceDefinitionV1CaBundle(required *apiextv1.CustomResourceDefinition, existing apiextv1.CustomResourceDefinition) {
+	if val, ok := existing.ObjectMeta.Annotations["service.beta.openshift.io/inject-cabundle"]; !ok || val != "true" {
```

Comment:
I think we should const the `"service.beta.openshift.io/inject-cabundle"` instead of inlining it and use the const also in tests etc.

Reply:
I agree, thanks for catching this!

I have created a new file `lib/resourcemerge/helper.go` where I have put the constant. I couldn't think of a more appropriate existing file or a better name for the file (maybe `lib/resourcemerge/constants.go`). If anybody has a better idea, please let me know.

Reply:
TBH I'd just place that in `apiext.go`, it's not used elsewhere. I'd only introduce a new file once we have more of these.

Reply:
Cool, if see a non-hypothetical use case for the const in the observable future, then let's keep the separate file :+1: